aries_vcx/src/handlers/proof_presentation/verifier/verifier.rs

```diff
@@ -45,52 +46,57 @@ impl Verifier {
 
     pub fn get_source_id(&self) -> String { self.verifier_sm.source_id() }
 
-    pub fn get_state(&self) -> VerifierState {
-        trace!("Verifier::get_state >>>");
-        self.verifier_sm.get_state()
-    }
-
-    pub fn presentation_status(&self) -> u32 {
-        trace!("Verifier::presentation_state >>>");
-        self.verifier_sm.presentation_status()
-    }
+    pub fn get_state(&self) -> VerifierState { self.verifier_sm.get_state() }
 
     pub fn handle_message(&mut self, message: VerifierMessages, send_message: Option<&impl Fn(&A2AMessage) -> VcxResult<()>>) -> VcxResult<()> {
         trace!("Verifier::handle_message >>> message: {:?}", message);
         self.step(message, send_message)
     }
 
-    pub fn send_presentation_request(&mut self, send_message: impl Fn(&A2AMessage) -> VcxResult<()>, comment: Option<String>) -> VcxResult<()> {
-        trace!("Verifier::send_presentation_request >>>");
-        self.step(VerifierMessages::SendPresentationRequest(comment), Some(&send_message))
+    pub fn send_presentation_request(&mut self, send_message: impl Fn(&A2AMessage) -> VcxResult<()>) -> VcxResult<()> {
+        if self.verifier_sm.get_state() == VerifierState::PresentationRequestSet {
+            let offer = self.verifier_sm.presentation_request()?.to_a2a_message();
+            send_message(&offer)?;
+            self.verifier_sm = self.verifier_sm.clone().mark_presentation_request_msg_sent()?;
+        }
+        Ok(())
     }
 
     pub fn send_ack(&mut self, send_message: impl Fn(&A2AMessage) -> VcxResult<()>) -> VcxResult<()> {
         trace!("Verifier::send_ack >>>");
         self.step(VerifierMessages::SendPresentationAck(), Some(&send_message))
     }
 
-    pub fn set_request(&mut self, presentation_request_data: PresentationRequestData) -> VcxResult<()> {
-        self.verifier_sm = self.verifier_sm.clone().set_request(presentation_request_data)?;
+    pub fn set_request(&mut self, presentation_request_data: PresentationRequestData, comment: Option<String>) -> VcxResult<()> {
+        trace!("Verifier::set_request >>> presentation_request_data: {:?}, comment: ${:?}", presentation_request_data, comment);
+        self.verifier_sm = self.verifier_sm.clone().set_request(&presentation_request_data, comment)?;
         Ok(())
     }
 
-    pub fn generate_presentation_request_msg(&self) -> VcxResult<String> {
```

Comment:
The name `generate_presentation_request_msg` was bit off, nothing was being generated. This function was renamed to `get_presentation_request_msg`